pkg/controller/service/service.go

```diff
@@ -13,25 +16,49 @@ func RenderServices(req router.Request, resp router.Response) error {
 	// reset, should get repopulated
 	svcInstance.Status.Endpoints = nil
 
-	objs, _, err := services.ToK8sService(req, svcInstance)
-	if err != nil {
-		return err
+	var svcList []*v1.ServiceInstance
+	http2Ports := ports.ByProtocol(svcInstance.Spec.Ports, true, v1.ProtocolHTTP2)
```

Comment:
Nit: Would you be able to use this function like 
```
http2Ports := ports.ByProtocol(svcInstance.Spec.Ports, v1.ProtocolHTTP2)
otherProtocol := []v1.Protocol{v1.ProtocolHTTP, v1.ProtocolTCP, v1.ProtocolUDP}
otherPorts := ports.ByProtocol(svcInstance.Spec.Ports, otherProtocols)
```
Making the function a bit easy to read

Reply:
I think my big issue with doing that is if we ever add a new protocol then this logic needs to get updated, whereas I'd rather have it naturally handle a situation like that

Reply:
yeah that's true